src/v/cluster/metadata_dissemination_types.h

```diff
@@ -179,33 +180,37 @@ struct get_leadership_request
 struct get_leadership_reply
   : serde::envelope<
       get_leadership_reply,
-      serde::version<0>,
+      serde::version<1>,
       serde::compat_version<0>> {
     using rpc_adl_exempt = std::true_type;
+    using is_success = ss::bool_class<struct glr_tag>;
     fragmented_vector<ntp_leader> leaders;
+    is_success success;
```

Comment:
What happens during upgrades? Does this default to false? So new versions getting replies from old versions will get `success = no`? Do we need to assign a default of `yes` here?

Reply:
good point, it should default to `is_success::yes`